Source/WebKit/UIProcess/mac/ViewGestureControllerMac.mm

```diff
@@ -119,6 +119,12 @@ @implementation WKSwipeCancellationTracker
 
     ASSERT(m_activeGestureType == ViewGestureType::None || m_activeGestureType == ViewGestureType::Magnification);
 
+    if (event.phase == NSEventPhaseBegan)
+        m_webPageProxy.pageClient().setCaretDecorationVisibility(false);
```

Comment:
Make `setCaretDecorationVisibility` a public method of `WebPageProxy` to avoid code duplication.